include/binlog/PrettyPrinter.cpp

```diff
@@ -90,6 +91,29 @@ void PrettyPrinter::printEvent(
   }
 }
 
+bool PrettyPrinter::printStruct(detail::OstreamBuffer& out, mserialize::Visitor::StructBegin sb, Range& input) const
+{
+  if (sb.name == "binlog::address" && sb.tag == "`value'L")
+  {
+    const std::uint64_t value = input.read<std::uint64_t>();
```

Comment:
Complete pedantic and I don't think we need to address this, but this implicitly limits binlog to only supporting architectures of 64 bits or lower. I don't know there any HPC's that run higher than 64 bits but RISC-V is future proofing for 128. https://en.wikipedia.org/wiki/128-bit_computing

Reply:
Good point. If I ever get the chance to port binlog to such a platform, I can extend support easily: the condition above checks the type of value: `L` stands for u64. For platforms like RISC-V, a different condition could match larger addresses.